src/PowerShell/Microsoft.WinGet.Client.Engine/Helpers/PackageManagerWrapper.cs

```diff
@@ -0,0 +1,134 @@
+﻿// -----------------------------------------------------------------------------
+// <copyright file="PackageManagerWrapper.cs" company="Microsoft Corporation">
+//     Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+// </copyright>
+// -----------------------------------------------------------------------------
+
+namespace Microsoft.WinGet.Client.Engine.Helpers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Runtime.InteropServices;
+    using Microsoft.Management.Deployment;
+    using Microsoft.WinGet.Client.Engine.Common;
+    using Windows.Foundation;
+
+    /// <summary>
+    /// Wrapper for PackageManager that handles rpc disconnections.
+    /// The object is disconnected when the server is killed or on an update.
+    /// </summary>
+    internal sealed class PackageManagerWrapper
+    {
+        private static readonly Lazy<PackageManagerWrapper> Lazy = new (() => new PackageManagerWrapper());
+
+        private PackageManager packageManager = null;
+
+        private PackageManagerWrapper()
+        {
+        }
+
+        /// <summary>
+        /// Gets the instance object.
+        /// </summary>
+        public static PackageManagerWrapper Instance
+        {
+            get { return Lazy.Value; }
+        }
+
+        /// <summary>
+        /// Wrapper for InstallPackageAsync.
+        /// </summary>
+        /// <param name="package">The package to install.</param>
+        /// <param name="options">The install options.</param>
+        /// <returns>An async operation with progress.</returns>
+        public IAsyncOperationWithProgress<InstallResult, InstallProgress> InstallPackageAsync(CatalogPackage package, InstallOptions options)
+        {
+            return this.ExecuteWithRetry(

```

Comment:
Any of these functions that take in previously created objects are not likely to succeed on a retry from a server process termination (since the objects they refer to are also now disconnected).  They should simply release the static reference and rethrow.

I'm not sure that the retry part is even worth it for the other cases, but you could keep it.